cmd/admin-heal-ops.go

```diff
@@ -692,18 +697,17 @@ func (h *healSequence) healBucket(bucket string) error {
 		return errServerNotInitialized
 	}
 
-	results, err := objectAPI.HealBucket(h.ctx, bucket, h.settings.DryRun, h.settings.Remove)
-	// push any available results before checking for error
-	for _, result := range results {
-		if perr := h.pushHealResultItem(result); perr != nil {
-			return perr
-		}
-	}
+	result, err := objectAPI.HealBucket(h.ctx, bucket, h.settings.DryRun, h.settings.Remove)
 	// handle heal-bucket error
 	if err != nil {
 		return err
 	}
 
+	// push any available results before checking for error
+	if err = h.pushHealResultItem(result); err != nil {
```

Comment:
It is not clear why this code is moved after `if err != nil { return err }` @harshavardhana, can you elaborate ?

Reply:
Because its not necessary anymore - previously we had errors and success together @vadmeste

Reply:
Can you remove this comment 'push any available results before checking for error' ?